### PR TITLE
ARC-1518-audience-tech-planning-create-recur-donor-counts-budget-with-intervals-of-month-and-day

### PR DESCRIPTION
**Pull request by ryantimjohn**
# Pull Request for dbt-arc-functions

- [x] Open the pull request as a draft so that you can run tests against the code and inspect the files in the Files view of the Pull Request interface in Github.

- [x] Once you're satisfied, convert the draft into an open Pull Request by clicking "Ready for Review" near the bottom of this page. You should not make any changes to the code except in dire emergency once you've converted from draft, so make sure you're really satisfied.

- [x] Add two of the principal contributors to this PR for review. Currently, the principal contributors are @dmbluestate , @Frydafly , and @ryantimjohn.

- [x] We've divided our PR templates into Macros, Utils and Docs PR requests. Delete the ones you're not using.
### Macros (delete if not using)

- [x] Link Jira tickets (or Github Issues) to PR here. Make sure that the Jira ticket has a good description as to why this PR is necessary and the business problem that this PR is solving. Moreover, if there have been any discussions of this ticket in Slack, make sure they're linked in the Jira ticket:

https://bluestate.atlassian.net/browse/ARC-1518

- [x] Make sure this is named using Jira extension so that naming convention is consistent. If it's not, that's okay. Go to the Jira ticket, click into the ticket, then on the right under Details, find Development, then click the Create branch button. Copy the Branch Name from there. Read here [how to rename a branch in Github](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-branches-in-your-repository/renaming-a-branch) to rename this branch. In the future, create branches from Jira.

- [x] Make sure the Macro is documented by running the notebook: `create_docs.ipynb` util in this repo. From there, you'll have to either fill in descriptions by hand OR use our `fill_in_descriptions_using_openai.ipynb` to fill in descriptions using OpenAI's ChatGPT. Always make sure to check over ChatGPT's column descriptions, as they're likely not perfect.

- [x] Test branch on a development branch of an existing client (ideally the one that raised the bug). Do this by changing revision of the package to the branch name in `packages.yml` file in dbt State client below

https://github.com/bsd/arc-uusa-dbt/pull/17

- [x] If new macro files were created or updated, re-run `create_or_update_standard_models.py` for the client dbt project, replacing models and dependencies

- [x] check in dbt that client package successfully runs `dbt deps`, `dbt compile`, and `dbt run`; screenshot this page

Screenshot 2023-09-19 at 2 59 45 PM

- [x] query the resulting staging model in bigquery (or by previewing mart in dbt cloud, or your IDE); screenshot this

Screenshot 2023-09-19 at 3 29 09 PM

### CODE REVIEW

File: macros/audience_budget/staging/stg_audience_budget_recur_donor_counts_with_interval_yearly.sql [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{% macro create_stg_audience_budget_recur_donor_counts_with_interval_yearly(
+    reference_name='stg_audience_budget_by_day') %}
+    select
+        last_day(date_day, year) as date_day,
+        'yearly' as interval_type,
+        donor_audience as donor_audience,
+        platform as join_source,
+        sum(total_donors_by_day) as recur_total_donor_count_budget,
+    from {{ ref(reference_name) }}
+    where donor_audience = 'recurring' or donor_audience = 'Monthly'
```
Comment (Frydafly) [inline macros/audience_budget/staging/stg_audience_budget_recur_donor_counts_with_interval_yearly.sql:10]: can we lower(donor_audience) = [name] ?

just to ensure more consistent matching when we vary our capitalization?

File: macros/audience_budget/staging/stg_audience_budget_onetime_donor_count_budget_monthly.sql [hunk 2]
```diff
@@ -9,7 +9,7 @@
                 sum(total_revenue_budget_by_day) as onetime_donor_count_budget,
                 sum(loyalty_new_donor_targets_by_day) as onetime_new_donor_count_budget
             from {{ ref("stg_audience_budget_by_day") }}
-            where donor_audience != 'recurring'
+            where donor_audience != 'recurring' and donor_audience != 'Monthly'
```
Comment (Frydafly) [inline macros/audience_budget/staging/stg_audience_budget_onetime_donor_count_budget_monthly.sql:12]: same here around using lower(donor_audience) when possible?

**Review by Frydafly — APPROVED** [review]
LGTM -- just pending some suggestions to improve matching

**Comment by ryantimjohn** [comment]
> LGTM -- just pending some suggestions to improve matching

@Frydafly, this make sense here! I think generally we'd be better off also just lowercasing everything early in the pipeline to make matches happen more easily. I can take on doing this for the spreadsheets if you think this is a good idea!

**Comment by ryantimjohn** [comment]
Screenshot 2023-09-21 at 11 38 46 AM

**Pull request merged**